src/tests/test_cmdline_runner.py

```diff
@@ -104,3 +108,223 @@ def test_invalid_cmdline_option(selenium):
+    assert result.stderr == ""
+
+
+@only_node
+def test_venv_pyodide_version(selenium, venv):
+    result = subprocess.run(
+        [
+            venv / "bin/python",
+            "-c",
+            "from pyodide import __version__; print(__version__)",
+        ],
+        capture_output=True,
+        encoding="utf8",
+    )
+    assert result.returncode == 0
+    assert result.stdout.strip() == pyodide.__version__
+    assert result.stderr == ""
+
+
+def install_pkg(venv, pkgname):
+    return subprocess.run(
+        [
+            venv / "bin/pip",
+            "install",
+            pkgname,
+        ],
+        capture_output=True,
+        encoding="utf8",
+    )
+
+
+def test_venv_pip_install_1(selenium, venv):
+    """pure Python package with no dependencies from pypi"""
+    result = install_pkg(venv, "more-itertools")
+    assert result.returncode == 0
+    cleaned_stdout = result.stdout.strip()
+    # delete lines indicating whether package was downloaded or used from cache
+    # since these don't reproduce.
+    cleaned_stdout = re.sub("^  .*?\n", "", cleaned_stdout, flags=re.MULTILINE)
+    assert (
+        cleaned_stdout
+        == dedent(
+            """
+            Looking in links: /src/dist
+            Collecting more-itertools
+            Installing collected packages: more-itertools
+            Successfully installed more-itertools-8.14.0
+            """
+        ).strip()
+    )
+
+    result = subprocess.run(
+        [
+            venv / "bin/python",
+            "-c",
+            dedent(
+                """
+                from more_itertools import chunked
+                iterable = range(9)
+                print(list(chunked(iterable, 3)))
+                """
+            ),
+        ],
+        capture_output=True,
+        encoding="utf-8",
+    )
+    assert result.returncode == 0
+    assert result.stdout == str([[0, 1, 2], [3, 4, 5], [6, 7, 8]]) + "\n"
+
+
+def test_venv_pip_install_2(selenium, venv):
+    """pure Python package with dependencies from pypi"""
+    result = install_pkg(venv, "requests")
+    assert result.returncode == 0
+    cleaned_stdout = result.stdout.strip()
+    # delete lines indicating whether package was downloaded or used from cache
+    # since these don't reproduce.
+    cleaned_stdout = re.sub("^  .*?\n", "", cleaned_stdout, flags=re.MULTILINE)
+    cleaned_stdout = "\n".join(sorted(cleaned_stdout.split("\n")))
+    assert (
+        cleaned_stdout
+        == dedent(
+            """
+            Collecting certifi>=2017.4.17
+            Collecting charset-normalizer<3,>=2
+            Collecting idna<4,>=2.5
+            Collecting requests
+            Collecting urllib3<1.27,>=1.21.1
+            Installing collected packages: urllib3, idna, charset-normalizer, certifi, requests
+            Looking in links: /src/dist
+            Successfully installed certifi-2022.6.15 charset-normalizer-2.1.1 idna-3.3 requests-2.28.1 urllib3-1.26.12
```

Comment:
This is probably bound to break often with version updates. Maybe we could replace any `([\d+]\.?)+` by `*` to make it more robust.